### PR TITLE
Leaderboard Calculation: Order Refactor \& Submission views formatting

### PR DESCRIPTION
**Pull request by nemaniarjun**
This commit refactors the Leaderboard Calculation Service to properly
handle the orders that are defined into the challenge.
Also refactors the submission detail and submission list views
to show the correct formatted value.

### CODE REVIEW

File: app/services/calculate_leaderboard_service.rb [hunk 1]
```diff
@@ -39,64 +36,16 @@ def call
...
+                      .order(created_at: :desc)
+                      .limit(1)
+                      .first
     window_border = most_recent.created_at - @round.ranking_window.hours
     return "'#{window_border.to_s(:db)}'"
   end
 
-  def get_order_by
-    challenge = @round.challenge
-    if (challenge.secondary_sort_order_cd.blank? || challenge.secondary_sort_order_cd == 'not_used')
```
Comment (skbly7) [inline app/services/calculate_leaderboard_service.rb]: What if blank need to be added back.

File: app/concepts/leaderboard/cells/table_row.rb [hunk 2]
```diff
@@ -12,13 +12,8 @@ def entry
     model
   end
 
-  def formatted_score
```
Comment (skbly7) [inline app/concepts/leaderboard/cells/table_row.rb]: `formatted_score` is used in baseline_row.erb

File: app/services/calculate_leaderboard_service.rb [hunk 3]
```diff
@@ -395,4 +355,33 @@ def set_leaderboard_sequences(leaderboard_type:)
...
+  end
+
+  def order_by(use_display: false)
+    challenge = @round.challenge
+
+    if challenge.latest_submission == true
+      return 'updated_at desc'
+    end
+
+    score_sort_order ||= sort_map(challenge.primary_sort_order_cd)
```
Comment (skbly7) [inline app/services/calculate_leaderboard_service.rb]: What will happen when “nil” is returned back (because of Not Used as value or so)?

Comment (nemaniarjun) [reply app/services/calculate_leaderboard_service.rb]: It can never be nil for primary score. New challenges have that as default.

**Pull request merged**